nbstripout/_utils.py

```diff
@@ -153,3 +154,97 @@ def strip_output(nb, keep_output, keep_count, extra_keys=[], drop_empty_cells=Fa
         for field in keys['cell']:
             pop_recursive(cell, field)
     return nb
+
+
+def process_pyproject_toml(toml_file_path):
+    """Extract config mapping from pyproject.toml file."""
+    try:
+        import tomllib  # python 3.11+
+    except ModuleNotFoundError:
+        import tomli as tomllib
+
+    with open(toml_file_path, 'rb') as f:
+        return tomllib.load(f).get('tool', {}).get('nbstripout', None)
+
+
+def process_setup_cfg(cfg_file_path):
+    """Extract config mapping from setup.cfg file."""
+    import configparser
+
+    reader = configparser.ConfigParser()
+    reader.read(cfg_file_path)
+    if not reader.has_section('nbstripout'):
+        return None
+
+    return reader['nbstripout']
+
+
+def merge_configuration_file(args):
+    """Merge configuration from a file into args."""
+    CONFIG_FILES = {
+        'pyproject.toml': process_pyproject_toml,
+        'setup.cfg': process_setup_cfg,
+    }
+    BOOL_TYPES = {
+        'yes': True,
+        'true': True,
+        'on': True,
+        'no': False,
+        'false': False,
+        'off': False
+    }
+
+    # Traverse the file tree common to all files given as argument looking for
+    # a configuration file
+    config_path = os.path.commonpath([os.path.abspath(file) for file in args.files]) if args.files else os.getcwd()
```

Comment:
Please use `pathlib` wherever possible.

Reply:
@kynan Thanks for the quick review. I believe I addressed all your comments except this one. I'm not too familiar with the `pathlib` module. Just did a quick search and couldn't find sth equivalent to `os.path.commonpath()`. Could you be more specific here?

Reply:
There's a whole interesting thread on S/O on commonpath -
https://stackoverflow.com/a/43613742

Though not nessacarily helpful - they do end up still using commonpath, and just wrapping it with Path.